configs/configupgrade/upgrade_body.go

```diff
@@ -873,3 +924,41 @@ var resourceTypeAutomaticConnectionExprs = map[string]map[string]string{
 		)`,
 	},
 }
+
+// copied directly from internal/initwd/getter.go
+var localSourcePrefixes = []string{
+	"./",
+	"../",
+	".\\",
+	"..\\",
+}
+
+func isMaybeRelativeLocalPath(addr string) bool {
+	for _, prefix := range localSourcePrefixes {
+		if strings.HasPrefix(addr, prefix) {
+			// it is _definitely_ a relative path
+			return false
+		}
+	}
+
+	_, err := regsrc.ParseModuleSource(addr)
+	if err == nil {
+		// it is a registry source
+		return false
+	}
+
+	realAddr, err := getter.Detect(addr, ".", getter.Detectors)
+	// it's unclear what the problem may be, so we'll
+	// just pass it through normally
+	if err != nil {
+		return false
+	} else {
+		if strings.HasPrefix(realAddr, "file://") {
+			_, err := os.Stat(realAddr[7:])
```

Comment:
I'm not sure if this'll do quite the right thing on Windows... Windows `file:` URLs can be kinda funny in situations where e.g. it's a UNC path, or where the drive letter is encoded, and so we might end up with extra slashes on the front of the path here.

Perhaps it doesn't matter when we're expecting the path to be relative anyway, but might be worth us testing this on Windows to make sure it doesn't do anything unfortunate.

Reply:
Oh good point, I just based this off the fact that "file://" is precisely what go-getter adds to the path (https://github.com/hashicorp/go-getter/blob/master/detect_file.go#L59). I will follow up on this one.